File: frontend/src/app/models/schedule/tag.ts
import { ActivityDTO } from "../activity/activity-dto";
import { ActivityColor } from "../enums/activity-color";
import { ScheduleTime } from "./schedule-time";

export interface Tag {
    id: number;
    userId: number;
    name: string;
    color: ActivityColor;
    activities: ActivityDTO[];
}

File: backend/NoPressure.DAL/Entities/Tag.cs
using NoPressure.Common.Enums;

namespace NoPressure.DAL.Entities
{
    public class Tag
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; }
        public string? Color { get; set; }
        public Team? Team { get; set; }
        public int? PlanId { get; set; }
        public List<Activity> Activities { get; set; }
    }
}
